LoopKitUI/PumpManagerUI.swift

```diff
@@ -9,37 +9,67 @@ import UIKit
 
+public struct PumpManagerSettings {
+    public var maxBasalRateUnitsPerHour: Double?
+    public var maxBolusUnits: Double?
+    public var basalSchedule: BasalRateSchedule?
+
+    public init(maxBasalRateUnitsPerHour: Double?, maxBolusUnits: Double?, basalSchedule: BasalRateSchedule?) {
+        self.maxBasalRateUnitsPerHour = maxBasalRateUnitsPerHour
+        self.maxBolusUnits = maxBolusUnits
+        self.basalSchedule = basalSchedule
+    }
+}
+
 public protocol PumpManagerUI: DeviceManagerUI, PumpManager, DeliveryLimitSettingsTableViewControllerSyncSource, BasalScheduleTableViewControllerSyncSource {
-    
-    // View for initial setup of device.
-    // If this method returns nil, it's expected that `init?(rawState: [:])` creates a non-nil manager
-    static func setupViewController(insulinTintColor: Color, guidanceColors: GuidanceColors) -> (UIViewController & PumpManagerSetupViewController & CompletionNotifying)?
+    /// Create and onboard a new pump manager.
+    ///
+    /// - Parameters:
+    ///     - settings: Settings used to configure the pump manager.
+    ///     - colorPalette: Color palette to use for any UI.
+    /// - Returns: Either a conforming view controller to create and onboard the pump manager, a newly created and onboarded pump manager, or an error.
+    static func setupViewController(initialSettings settings: PumpManagerSettings, colorPalette: LoopUIColorPalette) -> UIResult<UIViewController & PumpManagerCreateNotifying & PumpManagerOnboardNotifying & CompletionNotifying, PumpManagerUI, Error>
 
-    // View for managing device after initial setup
-    func settingsViewController(insulinTintColor: Color, guidanceColors: GuidanceColors) -> (UIViewController & CompletionNotifying)
+    /// Configure settings for an existing pump manager.
+    ///
+    /// - Parameters:
+    ///     - glucoseUnit: The glucose units to use.
+    ///     - colorPalette: Color palette to use for any UI.
+    /// - Returns: A view controller to configure an existing pump manager.
+    func settingsViewController(colorPalette: LoopUIColorPalette) -> (UIViewController & PumpManagerOnboardNotifying & CompletionNotifying)
 
     // View for recovering from delivery uncertainty
-    func deliveryUncertaintyRecoveryViewController(insulinTintColor: Color, guidanceColors: GuidanceColors) -> (UIViewController & CompletionNotifying)
+    func deliveryUncertaintyRecoveryViewController(colorPalette: LoopUIColorPalette) -> (UIViewController & CompletionNotifying)
 
     // Returns a class that can provide HUD views
-    func hudProvider(insulinTintColor: Color, guidanceColors: GuidanceColors) -> HUDProvider?
-    
+    func hudProvider(colorPalette: LoopUIColorPalette) -> HUDProvider?
+
     // Instantiates HUD view (typically reservoir volume) from the raw state returned by hudViewRawState
     static func createHUDView(rawValue: HUDProvider.HUDViewRawState) -> LevelHUDView?
 }
 
+public protocol PumpManagerCreateDelegate: AnyObject {
```

Comment:
Seems like a good simplification would be to combine `PumpManagerCreateDelegate` and `PumpManagerOnboardDelegate` into a  `PumpManagerSetupUIDelegate` that also conforms to `CompletionDelegate`. I don't think these protocols are being used independently during pumpmanager setup.
 
This would allow the parallel `*Notifying` types to be combine into a `PumpManagerSetupUINotifying`.

These changes would help with much of the "echoing code" like this:

```
            setupViewController.pumpManagerCreateDelegate = self
            setupViewController.pumpManagerOnboardDelegate = self
            setupViewController.completionDelegate = self
```

Reply:
`PumpManagerOnboardDelegate ` is used independently  in `settingsViewController`.

Per earlier conversations, the `setupViewController` can create a new pump manager, but leave it in an incomplete, not-yet-onboarded state (i.e. app force-quit in the middle of setup). When the app is restarted, it would note the existence of the pump manager and its not-yet-onboarded state and invoke `settingsViewController` to complete onboarding. Therefore, the `settingsViewController` needs to conform to the `PumpManagerOnboardNotifying` protocol and would need the `PumpManagerOnboardDelegate`.

Also, one problem with explicitly combining the delegates (particularly the pump manager delegates with the completion delegate) is that you cannot split delegate implementation. Right now `StatusTableViewController` has to act as a "switchboard" for pump manager create and onboard notification.

Reply:
Ok, I'm ok keeping them separate if they're used separately. we just have a proliferation of types here, multiplied by 3 (service/cgm/pump) and 2, (delegate/notifying).  Seems like a grouping makes sense, but I can understand if they're used separately, they need to stay separate.